src/model_params.jl

```diff
@@ -0,0 +1,295 @@
+mutable struct ModelParams
+    method::Type{<:AbstractBlockNLPSolver}
```

Comment:
Can't we specify directly?
```julia
method::AbstractBlockNLPSolver
```

Reply:
I tried that, but somehow doesn't work. It gives me this error:

```julia
ERROR: LoadError: MethodError: Cannot `convert` an object of type Type{BlockNLPAlgorithms.DualDecomposition} to an object of type BlockNLPAlgorithms.AbstractBlockNLPSolver
Closest candidates are:
  convert(::Type{T}, ::T) where T at /opt/julia-1.7.2/share/julia/base/essentials.jl:218
````